tests/MongoDB.Driver.Core.Tests/Core/Connections/BinaryConnectionTests.cs

```diff
@@ -276,6 +276,45 @@ public void Dispose_should_raise_the_correct_events()
             act.ShouldThrow<InvalidOperationException>();
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void ReceiveMessage_should_throw_error_when_message_is_an_invalid_size(
+            [Values(-1, 16000005)]
+            int length,
+            [Values(false, true)]
+            bool async)
+        {
+            using (var stream = new BlockingMemoryStream())
+            {
+                var bytes = BitConverter.GetBytes(length);
+                stream.Write(bytes, 0, bytes.Length);
+                stream.Seek(0, SeekOrigin.Begin);
+                var encoderSelector = new ReplyMessageEncoderSelector<BsonDocument>(BsonDocumentSerializer.Instance);
+
+                Action received;
```

Comment:
Hm, food for thought: as this is an `Action`, would a verb work better as a name than an adjective? Perhaps something like `ReceiveMessage`?

Reply:
I assume this was copy/pasted from an existing test method being adapted for the new test.

In brand new tests we would FluentAssertions Record.Exception instead.

Reply:
Ah true, I had missed this :)